### PR TITLE
add minimal inference workflow test

### PR DESCRIPTION
**Pull request by ahnitz**
In keeping with trying to have more of our workflows run in the CI, here is a minimal workflow for running pycbc inference. 

Depends on #3830

### CODE REVIEW

**Comment by ahnitz** [comment]
@spxiwh Ok, hopefully, this should be easier to see what is new now.

File: examples/search/check_job.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 time.sleep(30)
 while 1:
     time.sleep(5)
```
Comment (spxiwh) [inline examples/search/check_job.py:5]: Can we make the sleep time larger? 5s does make the log output pretty long ...

File: .github/workflows/inference-workflow.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        bash -e ../examples/search/submit.sh
+        ./status
+        python ../examples/search/check_job.py
+        find submitdir/work/ -type f -name '*.tar.gz' -delete
+    - name: store log files
+      if: always()
+      uses: actions/upload-artifact@v2
+      with:
+        name: logs
+        path: output/submitdir/work
```
Comment (spxiwh) [inline .github/workflows/inference-workflow.yml:51]: I think you called the output here gw_output.

**Review by spxiwh — APPROVED** [review]
Thanks for adding this. One minor thing that isn't working, and one comment, but nothing major.

File: .github/workflows/inference-workflow.yml (hunk 2 above)
Comment (spxiwh) [reply .github/workflows/inference-workflow.yml:51]: (To be clear, the logs aren't being uploaded, so that should be fixed, I think this is why(

**Pull request merged**